shared.tf

```diff
@@ -1,22 +1,30 @@
+resource "random_string" "random_resource_name_suffix" {
+  length  = 10
+  special = false
+  upper   = false
+}
+
 module "api_gateway" {
   source                = "./modules/api_gateway"
   global_tags           = var.global_tags
   tags                  = var.tags
   environment           = var.environment
   integration_type      = var.integration_type
-  resource_name_pattern = local.single_lambda_integration ? local.resource_name_pattern : "${local.resource_name_pattern}-frontend"
+  resource_name_pattern = local.resource_name_pattern
+  function_name         = local.function_name
   lambda_function_arn   = local.api_triggered_function_arn
 }
 
 module "secrets_manager" {
   count            = var.store_secret_in_secrets_manager ? 1 : 0
   integration_type = var.integration_type
   source           = "./modules/secrets_manager"
+  secrets_names    = local.default_secrets_names[var.integration_type]
 }
 
 module "lambda_role" {
   source                          = "./modules/role"
-  resource_name_pattern           = local.single_lambda_integration ? local.resource_name_pattern : "${local.resource_name_pattern}-frontend"
+  role_name                       = local.function_name
```

Comment:
Why is the role name getting the function name?

Reply:
Because this role and its policies were created explicitly for this lambda function, and with this function name there will be no doubt about why this role was created and who should use it